plugins/vbe/fu-plugin-vbe.h

```diff
@@ -0,0 +1,19 @@
+/*
+ * VBE plugin for fwupd,mmc-simple
+ *
+ * Copyright (C) 2018 Richard Hughes <richard@hughsie.com>
+ * Copyright (C) 2022 Google LLC
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#pragma once
+
+#include <fwupdplugin.h>
+
+enum { PROP_0, PROP_VBE_METHOD, PROP_VBE_FDT, PROP_VBE_NODE, PROP_LAST };
+
+typedef FuDevice *(*vbe_device_new_func)(FuContext *ctx,
+					 const gchar *vbe_method,
+					 const gchar *fdt,
+					 int node);
```

Comment:
I don't really like this, or the VbeDriver matching stuff. For other plugins, we create the correct FuDevice superclass directly -- and in this case there only seems to be one anyway, so it seems a ton of extra complexity for no real purpose.

Reply:
Well I need to start somewhere. I found that the docs in the readme are not visible, so I updated them.

Here is the main VBE idea. Every board potentially has its own way of doing updates, similar to how every USB device does. I am trying to implement common features, but I suspect there will be vendor-specific drivers too.

https://docs.google.com/document/d/e/2PACX-1vQjXLPWMIyVktaTMf8edHZYDrEvMYD_iNzIj1FgPmKF37fpglAC47Tt5cvPBC5fvTdoK-GA5Zv1wifo/pub